universal-application-tool-0.0.1/test/services/applicant/ApplicantDataTest.java

```diff
@@ -216,6 +216,54 @@ public void putList_writesNullIfListIsEmpty() {
         .isEqualTo("{\"applicant\":{\"favorite_fruits\":null},\"metadata\":{}}");
   }
 
+  @Test
+  public void putRepeatedEntities() {
+    ApplicantData data = new ApplicantData();
+    Path path = Path.create("applicant.children[1].pets[]");
+    ImmutableList<String> petNames = ImmutableList.of("bubbles", "luna", "taco");
+
+    data.putRepeatedEntities(path, petNames);
+
+    assertThat(data.asJsonString())
+        .isEqualTo(
+            "{\"applicant\":{\"children\":[{},{\"pets\":[{\"entity_name\":\"bubbles\"},{\"entity_name\":\"luna\"},{\"entity_name\":\"taco\"}]}]},\"metadata\":{}}");
+  }
+
+  @Test
+  public void putRepeatedEntities_withPrexistingData() {
+    ApplicantData data =
+        new ApplicantData(
+            "{\"applicant\":{\"children\":[{},{\"entity_name\":\"an old name\",\"pets\":["
```

Comment:
question: why is there no index 0 entity?

Reply:
ah so this renames the existing entity_names? What if we want to add more?

Reply:
this test checks that renames work without changing other parts of the repeated entity (e.g. pets).  The other test tests adding new ones.